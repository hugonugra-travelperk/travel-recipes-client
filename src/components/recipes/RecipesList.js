import React, { useState, useEffect } from "react";
import useAxios from '../../hooks/useAxios'
import RecipeElement from "./RecipeElement"
import SearchBox from "./SearchBox"

export default function RecipesList(){
  const base_path = '/recipes/'
  const [searchTerm, setSearchTerm] = useState('')

  const { dispatch, response, loading, error } = useAxios({
    method: 'GET',
    url: searchTerm ? base_path + '?name=' + searchTerm : base_path,
  }, () => {})

  useEffect(() => {
    dispatch();
  }, [searchTerm]);

  function updateSearchTerm(term) {
    setSearchTerm(term);
  }

  return(
    <>
      <SearchBox
        updateSearchTerm={updateSearchTerm}
      />
      {loading && <div>Loading...</div>}
      {error && <div>{error.message}</div>}
      {response && response.map(function(item){
        return (
          <RecipeElement
            detail={false}
            key={item.id}
            recipe={item}
          />
        )
      })}
    </>
    
  )
}
