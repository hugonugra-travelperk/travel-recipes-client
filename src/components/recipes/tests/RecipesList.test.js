import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";
import RecipesList from "../RecipesList"

afterEach(cleanup);

it("renders", () => {
  const {asFragment} = render(
    <BrowserRouter>
      <RecipesList
        handleUpdate={()=>{}} recipes = {[{'id':'1', 'name': 'test text recipe',
          'ingredients': [{'id': '1', 'name': 'test text ingredient'}]}]
        }
      />
    </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
