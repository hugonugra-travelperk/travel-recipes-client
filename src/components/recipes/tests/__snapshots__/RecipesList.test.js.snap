// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div>
    <input
      class="sc-pVTFL edDqrh"
      placeholder="Search by Name"
      type="text"
      value=""
    />
    <button
      class="sc-iCfMLu gMEgxH"
    >
       Search 
    </button>
  </div>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
